src/pages/StatePartner Dashboard/index.js

```diff
@@ -0,0 +1,157 @@
+  const [districtPartner, setDistrictPartner] = useState([]);
+  const [districtPartnerData, setDistrictPartnerData] = useState([]);
+  const [filterData, setFilterData] = useState([]);
+  const [value, setValue] = useState();
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/partners/23/groups`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      setDistrictPartner(res.data);
+      setDistrictPartnerData(res.data.partner_groups_data);
+      setFilterData([...filterData, res.data.partner_groups_data[0]]);
+    });
+  }, []);
+
+  const handleSelcet = (e) => {
+    const stateFilterData = districtPartnerData.filter(
+      (el) => el.partner_group_name == e.target.value
+    );
+    setFilterData(stateFilterData);
+    setValue(e.target.value);
+  };
+
+  const canSpecifyPartnerGroupId =
+    hasOneFrom(user.data.user.rolesList, [
+      "admin",
+      "partner",
+      "partner_view",
+    ]) && user.data.user.partner_group_id;
+
+  if (canSpecifyPartnerGroupId) {
+    return (
+      <div className="state-partner-dashboard">
+        {/* <h2> Hariyana Partnership dashboard</h2> */}
+        <h2>{districtPartner.state_name} Dashboard</h2>
+        <h4>Overview</h4>
+        <div className="state-partner-overview">
+          <div className="state-partner-overview-card">
+            <span className="overview-card-number">
+              {districtPartner.total_no_of_groups}
+            </span>
+            <span className="overview-card-label">Total Number of Groups</span>
+          </div>
+          <div className="state-partner-overview-card">
+            <span className="overview-card-number">
+              {districtPartner.total_no_of_partners}
+            </span>
+            <span className="overview-card-label">
+              Total Number of Partners
+            </span>
+          </div>
+          <div className="state-partner-overview-card">
+            <span className="overview-card-number">
+              {districtPartner.total_no_of_students}
+            </span>
+            <span className="overview-card-label">Total Enrolled Students</span>
+          </div>
+        </div>
+        <div className="state-partner-choose-district">
+          <h4>Please choose a group</h4>
+          <select
+            className="input-for-district"
+            onChange={handleSelcet}
+            value={value}
+          >
+            <option value="all">All</option>
+            {districtPartnerData.map((item, i) => {
+              return (
+                <>
+                  <option
+                    key={i}
+                    value={item.partner_group_name}
+                    selected={i === 0 ? "selected" : ""}
+                  >
+                    {item.partner_group_name}
+                  </option>
+                </>
+              );
```

Comment:
The outer `<>` is not necessary here because the JSX you're returning only has one root element `<option>`.  You would need to use a JSX fragment if the `return` statement included adjacent JSX elements such as:
```
return (
  <option>Something</option>
  <option>Something else</option>
);
```
Also, to conditionally select an option using a `selected` attribute, you'd pass it `true` or `false`.  `"selected"` and `""` work because the first value `"selected"` is truthy and the second one of the empty string `""` is falsy but you could just write `i === 0` instead.  (Sorry about my mistake before when I just wrote `{i === 0 ? "selected" : ""}` which won't work.)

One more note is that using `selected` is generally not recommended (see: https://stackoverflow.com/a/21736116), but I see you're using it here because the list of district options will be empty until the district partner data is loaded. 

```suggestion
              return (
                  <option
                    key={i}
                    value={item.partner_group_name}
                    selected={i === 0}
                  >
                    {item.partner_group_name}
                  </option>
              );
```

Reply:
We are using selected to display the second option. As `All` is the first option, by default it gets displayed but we want to display the 1st element of the district.